### PR TITLE
Added support for Vue + Typescript and added eslint + prettier

### PR DESCRIPTION
**Pull request by patrick11514**
Added support for using Vue and Typescript in frontend. 
Added Prettier + Eslint for linting and formatting frontend files.

### CODE REVIEW

File: frontend/rollup.config.js [hunk 1]
```diff
@@ -43,6 +46,19 @@ export default {
             emitCss: true,
 		}),
 
+		vue({
+            preprocessStyles: true,
+        }),
+        // Replace the environment variables - https://github.com/rollup/rollup/issues/805
+        // btw the vue/dist/vue.esm.browser didn't work
+        replace({
+            'process.env.NODE_ENV': JSON.stringify('development'),
```
Comment (Kobzol) [inline frontend/rollup.config.js:55]: This should be `"production"` when `production` is true, right? It does seem to save ~30 KiB from the resulting JS archive.

**Review by Kobzol — COMMENTED** [review]
Thank you! I will probably make a bunch of changes to the TS config in a follow-up PR where I'll configure it on CI, but that does not need to block this PR.

Left one nit, otherwise LGTM.

File: frontend/rollup.config.js (hunk 1 above)
Comment (patrick11514) [reply frontend/rollup.config.js:55]: Yes, it should by production, my bad, also it can be dynamic based on production value.

**Comment by geordi** [comment]
I'm a bit sad that [my branch](https://github.com/mrlvsb/kelvin/tree/vuejs) was kind of ignored, but integration into existing Rollup is nice.

**Comment by patrick11514** [comment]
> I'm a bit sad that [my branch](https://github.com/mrlvsb/kelvin/tree/vuejs) was kind of ignored, but integration into existing Rollup is nice.

Actually I was using your branch, but then we decided with @Kobzol to implement it in current Rollup setup next to Svelte for smoother transition from Svelte to Vue web components.

**Comment by geordi** [comment]
> > I'm a bit sad that [my branch](https://github.com/mrlvsb/kelvin/tree/vuejs) was kind of ignored, but integration into existing Rollup is nice.
> 
> Actually I was using your branch, but then we decided with @Kobzol to implement it in current Rollup setup next to Svelte for smoother transition from Svelte to Vue web components.

OK, no problem at all and thank you for the clarification. It's definitely easier to do transition one technology at time so we don't break things.

I'm VERY happy we can move forward with Vue integration! Thanks!

**Comment by Kobzol** [comment]
The production bundle was only increased by ~30 KiB in production mode, that's quite nice. And today is Friday, so what better time to ship stuff :)

**Review by Kobzol — APPROVED** [review]

**Pull request merged**